provider/internal/device/asa/resource.go

```diff
@@ -305,14 +299,20 @@ func (r *AsaDeviceResource) Update(ctx context.Context, req resource.UpdateReque
 		return
 	}
 
-	tagsInp := tags.New(util.TFStringListToGoStringList(planData.Labels)...)
+	// convert tf tags to go tags
+	tagsGoList, err := util.TFStringSetToGoStringList(ctx, planData.Labels)
+	if err != nil {
+		res.Diagnostics.AddError("error while converting terraform tags to go tags", err.Error())
+		return
+	}
+	tags_ := tags.New(tagsGoList...)
```

Comment:
Yea I cant think of a nice short name since we are using a package named `tags` here, but anyway, I inline it instead so that the name is not longer needed